### PR TITLE
Adding support to Distributed Configuration via Google Cloud Secret Manager

### PR DESCRIPTION
**Pull request by viniciusccarvalho**
Long overdue, life got in the way.

Looking forward for reviews and improvements

Fixed a few typos on the GCP docs as well

Some of the gcp-common beans had to be annotated with `@BootstrapContextCompatible`

### CODE REVIEW

**Comment by viniciusccarvalho** [comment]
I can't tell why the check fails on the runner, and not on my machine. I think I ran the same build `./gradlew check`, it looks like the checkstyle rules are different and way more strict than the ones on the project.

For instance:

```
Error: eckstyle] [ERROR] /home/runner/work/micronaut-gcp/micronaut-gcp/gcp-logging/src/main/java/io/micronaut/gcp/logging/GoogleCloudPropertyDefiner.java:41:61: http:// URLs are not allowed but got 'http://metadata.google.internal'. Use https:// instead. [NoHttp]```

That is a string declaration for the google cloud metadata server that happens to not be available on `https` seems to be a very strict checkstyle rule.

Let me know what needs to be done for the tests to pass.

Cheers

**Comment by alvarosanchez** [comment]
@viniciusccarvalho you can disable checkstyle on a per-line basis wrapping them with

```java
//CHECKSTYLE:OFF
...
//CHECKSTYLE:ON
```

**Comment by viniciusccarvalho** [comment]
Thanks @alvarosanchez I'm still intrigued by two things:

1. The offending class is on the logging module, not part of this PR, did the checkstyle rules changed?
2. On my machine the build passes with no warning, I have forced checkstyleMain, check, and they all pass with no warning. Still not clear where the checkstyle warnings are coming from.

**Comment by alvarosanchez** [comment]
It is actually my fault, when merging #315 I inadvertently reverted the change to use only Java 11.

I'm so sorry 🙏 

I'll fix it in master and rebase your PR

**Comment by alvarosanchez** [comment]
Ok, now it works. The actions failure is because of https://github.com/micronaut-projects/micronaut-project-template/issues/62, and can be ignored.

Apart from that, is this ready for review?

**Comment by viniciusccarvalho** [comment]
It should be thank you!

File: gcp-secret-manager/src/main/resources/logback.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+             ch.qos.logback.classic.encoder.PatternLayoutEncoder by default -->
+        <encoder>
+            <pattern>%cyan(%d{HH:mm:ss.SSS}) %gray([%thread]) %highlight(%-5level) %magenta(%logger{36}) - %msg%n</pattern>
+        </encoder>
+    </appender>
+    <logger name="com.google.cloud.secretmanager" level="DEBUG"/>
+    <root level="info">
+        <appender-ref ref="STDOUT" />
+    </root>
+</configuration>
```
Comment (graemerocher) [inline gcp-secret-manager/src/main/resources/logback.xml:15]: This shouldn't be in `src/main/resources`. Maybe you meant `src/test/resourses`?

File: gcp-secret-manager/src/main/java/io/micronaut/gcp/secretmanager/configuration/SecretManagerConfigurationProperties.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * @since 3.2.0
+ */
+@ConfigurationProperties(SecretManagerConfigurationProperties.PREFIX)
+@BootstrapContextCompatible
+public class SecretManagerConfigurationProperties {
+    public static final String PREFIX = GoogleCloudConfiguration.PREFIX + ".secret-manager";
+
+    private Set<String> customConfigs = new HashSet<>();
+    private Set<String> keys = new HashSet<>();
+
```
Comment (graemerocher) [inline gcp-secret-manager/src/main/java/io/micronaut/gcp/secretmanager/configuration/SecretManagerConfigurationProperties.java:37]: Does order not matter for these configs?

**Review by graemerocher — CHANGES_REQUESTED** [review]
Just a couple of minor things. Nice work!

**Review by graemerocher — APPROVED** [review]

**Comment by graemerocher** [comment]
@ilopmar @alvarosanchez when this is merged and the BOM updated for 2.3.x please add an entry in the what's new docs crediting @viniciusccarvalho work for 2.3.x

Thanks @viniciusccarvalho !

**Comment by alvarosanchez** [comment]
Thank you very much @viniciusccarvalho! I have in fact invited you as a collaborator to this repo, so no more need to work on your fork.

**Comment by viniciusccarvalho** [comment]
Thank you folks, it's been an amazing journey being able to contribute to this amazing framework. Truly humbled to be learn more every day from you all :)

**Pull request merged**